### ISSUE TITLE
Dual list selector 

### ISSUE THREAD

**Issue by LHinson** [patternfly/patternfly-design#764]
http://www.patternfly.org/pattern-library/forms-and-controls/dual-list-selector/

**Comment by nlcwong** [comment]
@mcarrano @LHinson 

The cost management application in the cloud.redhat.com product would like to have this component available in PF4.  Here is a screenshot of how we are using it (PF3 version) temporarily.

![Screen Shot 2020-04-08 at 4 48 35 PM](https://user-images.githubusercontent.com/31672086/78835553-6268bd80-79be-11ea-8f4b-128a6ea05fdd.png)

I will update that this section with a document that will outline our use case and mocks going forward.

**Comment by bdiring** [comment]
+1 to @nlcwong's comment.

I'm starting designs for a "Create project" wizard for the new Migration Toolkit for Applications (currently RHAMT). The existing UI uses a similar component, so I'm just trying to update it. Here are screenshots of the current design and a first pass, low-fidelity mock of what I've put together so far.

![Current RHAMT UI](https://user-images.githubusercontent.com/54444091/81006883-3c4c0700-8e1e-11ea-86cb-d09183102f8d.png)

![Low-fidelity mock](https://user-images.githubusercontent.com/54444091/81006902-41a95180-8e1e-11ea-968c-b38a7f25d33c.png)

**Comment by nlcwong** [comment]
Design doc: https://docs.google.com/document/d/1HMlBDpK8VO2O8YWdnIdKiyiYiy7tFVUnDwrrvTp8Lk4/edit

**Comment by mmenestr** [comment]
Here are [the mocks](https://marvelapp.com/prototype/12e47bhj) to this issue. Based on the PF meetings, the design is close to finalized - just need to tweak a few things, and will update this issue once i'm done. Also, based on the PF meeting, it seems like the preferred mobile layout was the toggle view ([here is the poll link](https://www.strawpoll.me/20988264/r)).

**Comment by mcarrano** [comment]
Have opened https://github.com/patternfly/patternfly/issues/3556 to begin implementation.

**Comment by mmenestr** [comment]
@mcarrano after talking with @mcoker and Katie from the react team, I decided to make the dual list have its own custom list implementation instead of using the tree view. The reason being, I wanted to tweak multiple things about the tree view component, including the way a selection is shown, the spacing, the font size and the badge positioning. Despite this, implementation should still be relatively smooth as the tree view component can be used as a base.

Here are the final mocks: https://marvelapp.com/project/5163382 - these include spec and explanation "slides" as well as demos of the way the component is meant to work!

Here is also the initial design document that I tweaked to match the mocks  - https://docs.google.com/document/d/1HMlBDpK8VO2O8YWdnIdKiyiYiy7tFVUnDwrrvTp8Lk4/edit#

This should cover it, but let me know if there's anything missing/if you have any concerns!
